.cicd/platforms/unpinned/optionals-NA-unpinned.dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM debian:10
+ENV VERSION 1
+
+# minimum packages to complete a build and run standard unit test suite
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y build-essential cmake curl git jq libboost-dev \
+                       libboost-chrono-dev libboost-date-time-dev libboost-filesystem-dev libboost-iostreams-dev \
+                       libboost-program-options-dev libboost-regex-dev libboost-test-dev \
+                       libcurl4-gnutls-dev libgmp-dev libssl-dev \
+                       libusb-1.0-0-dev libz-dev llvm-dev pkg-config python3 python3-requests
+
+# libpq: blockvault support
+RUN apt-get install -y libpq-dev postgresql-server-dev-all
+# postgresql: blockvault unit tests
+RUN apt-get install -y postgresql
+# nodejs & npm: ship unit test
+RUN apt-get install -y nodejs npm
```

Comment:
Aptitude leaves a bunch of temporary files and other garbage behind that causes the size of the layers in your docker containers to bloat. Docker recommends[1,2] cleaning up package manager cache to prevent this. [We do this](https://github.com/EOSIO/eos/blob/607047d0849c6f22bb03958d43a99641bf666d82/.cicd/platforms/pinned/ubuntu-20.04-pinned.dockerfile#L12) by appending this string to all of our `RUN` commands invoking `apt`:
```docker
 && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
```
Unfortunately, this means `apt-get update` is also required in each `RUN` command, but it ultimately results in smaller containers. So, your `RUN` commands would become:
```dockerfile
# minimum packages to complete a build and run standard unit test suite
RUN apt-get update && \
    apt-get upgrade -y && \
    apt-get install -y build-essential cmake curl git jq libboost-dev \
                       libboost-chrono-dev libboost-date-time-dev libboost-filesystem-dev libboost-iostreams-dev \
                       libboost-program-options-dev libboost-regex-dev libboost-test-dev \
                       libcurl4-gnutls-dev libgmp-dev libssl-dev \
                       libusb-1.0-0-dev libz-dev llvm-dev pkg-config python3 python3-requests && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*

# libpq: blockvault support
RUN apt-get update && \
    apt-get install -y libpq-dev postgresql-server-dev-all && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
# postgresql: blockvault unit tests
RUN apt-get update && \
    apt-get install -y postgresql && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
# nodejs & npm: ship unit test
RUN apt-get update && \
    apt-get install -y nodejs npm && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
```


**Links**
1. Intro Guide to Dockerfile Best Practices > Tip Number 5
   https://www.docker.com/blog/intro-guide-to-dockerfile-best-practices
1. Best practices for writing Dockerfiles > RUN > apt-get
   https://docs.docker.com/develop/develop-images/dockerfile_best-practices

Reply:
Also, for long package lists, I like to indent them one row and put them on their own line, like [this](https://github.com/EOSIO/auto-eks-sync-nodes/blob/5ae53c5cd03c529142c93dadeaace9abc7777d92/snapshot/dockerfile#L3):
```dockerfile
RUN apt-get update && \
    apt-get upgrade -y && \
    apt-get install -y
        build-essential
        cmake
        curl
        git
        jq
        libboost-chrono-dev
        libboost-date-time-dev
        libboost-dev
        libboost-filesystem-dev
        libboost-iostreams-dev
        libboost-program-options-dev
        libboost-regex-dev
        libboost-test-dev
        libcurl4-gnutls-dev
        libgmp-dev
        libssl-dev
        libusb-1.0-0-dev
        libz-dev
        llvm-dev
        pkg-config
        python3
        python3-requests && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
```
It makes it super easy to scan over the package list looking for something, preserves line history for package changes, and it makes the file look really clean. Completely up to you, though, just a suggestion. I used this transform to sort the list of packages in my clipboard:
```bash
pbpaste | tr -d '\\\n' | sed -E 's_\s+_ _g' | tr ' ' '\n' | sort | pbcopy
```

Reply:
That explains why, in my testing, the `apt-get clean` line always looked like a `no-op`! Interesting, thanks. I still think we should remove the package lists in all of our dockerfiles, but 1% is pretty negligible.